app/templates/components/public/ticket-list.hbs

```diff
@@ -58,10 +58,30 @@
         </tr>
       </tfoot>
     </table>
-    <div class="ui row grid">
-      <div class="column right aligned">
-        <a href="#">{{t 'Enter promotional code'}}</a>
-      </div>
+    <div class="ui grid">
+      {{#if enterPromotionalCode}}
+        <div class="ui row">
+          <div class="{{if device.isBiggerThanTablet 'right floated eight wide'}} column">
+            <div class="field">
+              <div class="ui action fluid input">
+                {{input type='text' name='promotional_code'}}
+                <div role='button' class="ui icon button" {{action 'applyPromotionalCode'}}>
+                  <i class="checkmark icon"></i>
+                </div>
+                <div role='button' class="ui black icon button" {{action 'togglePromotionalCode'}}>
```

Comment:
@anu-007 this is not a negative action. It is a cancellation action. Please see all the modals for reference. By clicking that icon user is not deleting the promotional code. He is cancelling the choice to enter it. Hence the black colour.